src/sentry/tasks/integrations/slack/find_channel_id_for_rule.py

```diff
@@ -78,18 +74,11 @@ def find_channel_id_for_rule(
     # endpoints but need some time limit imposed. 3 minutes should be more than enough time,
     # we can always update later
     try:
-        if features.has("organizations:slack-sdk-get-channel-id", organization):
-            channel_data = get_channel_id_with_timeout(
-                integration,
-                channel_name,
-                3 * 60,
-            )
-        else:
-            channel_data = get_channel_id_with_timeout_deprecated(
-                integration,
-                channel_name,
-                3 * 60,
-            )
```

Comment:
Removing the deprecated path is a bigger change than just updating tests, so we should update the PR title to reflect that (so as to have it in the commit log when you eventually merge).